tests/cupy_tests/manipulation_tests/test_kind.py

```diff
@@ -10,6 +10,19 @@
 @testing.gpu
 class TestKind(unittest.TestCase):
 
+    @testing.for_all_dtypes()
+    def test_asarray_chkfinite(self, dtype):
+        a = [0, 4, 0, 5]
+        a_gpu = cupy.asarray_chkfinite(a, dtype=dtype)
+        a_cpu = numpy.asarray_chkfinite(a, dtype=dtype)
+        assert a_gpu.dtype == a_cpu.dtype
```

Comment:
Using the testing helpers (see above) allows you to write a lot less boilerplate codes and increase test coverages (after the return a lot of things are checked).
```suggestion
        return xp.asarray_chkfinite(a, dtype=dtype, order=order)
```

Reply:
Got it! It's cooler than the previous approach.